File: ruby/lib/enolib/elements/list_item.rb
# frozen_string_literal: true

module Enolib
  class ListItem < ValueElementBase
    def parent
      @parent || List.new(@context, @instruction[:parent])
    end

    def to_s
      "#<Enolib::ListItem value=#{print_value}>"
    end
  end
end

File: python/enolib/elements/list_item.py
from . import list as list_module  # don't globally override built-in list function
from .value_element_base import ValueElementBase


class ListItem(ValueElementBase):
    def __repr__(self):
        return f"<class ListItem value={self._print_value()}>"

    def parent(self):
        return self._parent or list_module.List(self._context, self._instruction.parent)
